BudgetOptimizer raises Pydantic deprecation warnings re: `arbitrary_types_allowed` in `Field()`

I've got some code that imports/uses `geometric_adstock` in my own custom MMM model. When importing this it goes through the usual module imports etc. and raises some warnings when it gets to https://github.com/pymc-labs/pymc-marketing/blob/355928335ed31c2e860d301dd41c53646e2ecfbf/pymc_marketing/mmm/budget_optimizer.py#L641

```
src/mmm_v2/model/ModelBuilder.py:22: in <module>
    from pymc_marketing.mmm.transformers import geometric_adstock
/opt/venv/lib/python3.12/site-packages/pymc_marketing/__init__.py:21: in <module>
    from pymc_marketing import bass, clv, customer_choice, mmm
/opt/venv/lib/python3.12/site-packages/pymc_marketing/customer_choice/__init__.py:16: in <module>
    from pymc_marketing.customer_choice.mv_its import (
/opt/venv/lib/python3.12/site-packages/pymc_marketing/customer_choice/mv_its.py:29: in <module>
    from pymc_marketing.mmm.additive_effect import MuEffect
/opt/venv/lib/python3.12/site-packages/pymc_marketing/mmm/__init__.py:67: in <module>
    from pymc_marketing.mmm.time_slice_cross_validation import (
/opt/venv/lib/python3.12/site-packages/pymc_marketing/mmm/time_slice_cross_validation.py:32: in <module>
    from pymc_marketing.mmm.builders.yaml import build_mmm_from_yaml
/opt/venv/lib/python3.12/site-packages/pymc_marketing/mmm/builders/__init__.py:17: in <module>
    from pymc_marketing.mmm.builders.yaml import build_mmm_from_yaml
/opt/venv/lib/python3.12/site-packages/pymc_marketing/mmm/builders/yaml.py:28: in <module>
    from pymc_marketing.mmm.multidimensional import MMM
/opt/venv/lib/python3.12/site-packages/pymc_marketing/mmm/multidimensional.py:180: in <module>
    from pymc_marketing.mmm.budget_optimizer import OptimizerCompatibleModelWrapper
/opt/venv/lib/python3.12/site-packages/pymc_marketing/mmm/budget_optimizer.py:562: in <module>
    class BudgetOptimizer(BaseModel):
/opt/venv/lib/python3.12/site-packages/pymc_marketing/mmm/budget_optimizer.py:602: in BudgetOptimizer
    mmm_model: InstanceOf[OptimizerCompatibleModelWrapper] = Field(
/opt/venv/lib/python3.12/site-packages/pydantic/fields.py:1325: in Field
    warn(
E   pydantic.warnings.PydanticDeprecatedSince20: Using extra keyword arguments on `Field` is deprecated and will be removed. Use `json_schema_extra` instead. (Extra keys: 'arbitrary_types_allowed'). Deprecated in Pydantic V2.0 to be removed in V3.0. See Pydantic V2 Migration Guide at https://errors.pydantic.dev/2.12/migration/
```

`arbitrary_types_allowed` is (correctly) used in the model config dictionary, but it is not a kwarg that does anything specific when used inside `Field()` annotations. The usage of this in `budget_optimizer.py` is raising warnings about extra kwargs in `Field`, but realistically I think it can just be removed?

I've prepared a PR that removes its use across `Field` annotations in the library, and took the liberty of cleaning up uses of the `arbitrary_types_allowed` kwarg as config values (mostly in `validate_call`) by throwing them into `ConfigDict`s

--- a/pymc_marketing/mmm/budget_optimizer.py
+++ b/pymc_marketing/mmm/budget_optimizer.py
@@ -641,7 +641,6 @@ class BudgetOptimizer(BaseModel):
     mmm_model: InstanceOf[OptimizerCompatibleModelWrapper] = Field(
         ...,
         description="The marketing mix model to optimize.",
-        arbitrary_types_allowed=True,
         alias="model",
     )
 
@@ -653,7 +652,6 @@ class BudgetOptimizer(BaseModel):
     utility_function: UtilityFunctionType = Field(
         default=average_response,
         description="Utility function to maximize.",
-        arbitrary_types_allowed=True,
     )
 
     budgets_to_optimize: DataArray | None = Field(
@@ -664,7 +662,6 @@ class BudgetOptimizer(BaseModel):
     custom_constraints: Sequence[Constraint] = Field(
         default=(),
         description="Custom constraints for the optimizer.",
-        arbitrary_types_allowed=True,
     )
 
     default_constraints: bool = Field(

--- a/pymc_marketing/mmm/causal.py
+++ b/pymc_marketing/mmm/causal.py
@@ -31,7 +31,7 @@ import pandas as pd
 import pymc as pm
 import pytensor
 import pytensor.tensor as pt
-from pydantic import Field, InstanceOf, validate_call
+from pydantic import ConfigDict, Field, InstanceOf, validate_call
 from pymc_extras.prior import Prior
 
 try:
@@ -654,7 +654,7 @@ class TBFPC:
     - Kass, R. & Raftery, A. (1995). "Bayes Factors."
     """
 
-    @validate_call(config=dict(arbitrary_types_allowed=True))
+    @validate_call(config=ConfigDict(arbitrary_types_allowed=True))
     def __init__(
         self,
         target: Annotated[

--- a/pymc_marketing/mmm/multidimensional.py
+++ b/pymc_marketing/mmm/multidimensional.py
@@ -162,7 +162,7 @@ import pymc as pm
 import pymc.dims as pmd
 import pytensor.xtensor as ptx
 import xarray as xr
-from pydantic import Field, InstanceOf, StrictBool, validate_call
+from pydantic import ConfigDict, Field, InstanceOf, StrictBool, validate_call
 from pymc.model.fgraph import clone_model as cm
 from pymc.util import RandomState
 from pymc_extras.deserialize import deserialize
@@ -2348,7 +2348,7 @@ class MMM(RegressionModelBuilder):
 
         return posterior_predictive_samples
 
-    @validate_call(config={"arbitrary_types_allowed": True})
+    @validate_call(config=ConfigDict(arbitrary_types_allowed=True))
     def sample_saturation_curve(
         self,
         max_value: float = Field(
@@ -2506,7 +2506,7 @@ class MMM(RegressionModelBuilder):
 
         return curve
 
-    @validate_call(config={"arbitrary_types_allowed": True})
+    @validate_call(config=ConfigDict(arbitrary_types_allowed=True))
     def sample_adstock_curve(
         self,
         amount: float = Field(

